docs: add create response calculator section (#700)
Layr-Labs/eigensdk-go: operator/README.md

- [Incredible Dot Product](https://github.com/Layr-Labs/eigensdk-go/blob/v2-dev-1/examples/incredible-dot-product/operator/main.go)
- [Awesome Vault Service](https://github.com/Layr-Labs/eigensdk-go/blob/v2-dev-1/examples/awesome-vault-service/operator/main.go)

## How to create your Response Calculator

To create your Response Calculator you have to declare a struct that satisfies the `ResponseCalculator` interface:

``` go
   type ResponseCalculator[Input any, Output any] interface {
      ComputeResponse(taskIndex uint32, input Input) (Output, error)
   }
```

The `ResponseCalculator` interface has the `ComputeResponse` method, which computes the response of a task given its index and input, returning the output, or an error if the computation fails.

> We recommend implementing your own reponse calculator if you need to save state between responses. If you need to save a state between the operator responses, then you should implement your Response calculator following the above interface. As a suggestion, you should place the calculation logic in the `ComputeResponse` method and initialize the necessary things for calculating the response on the new response calculator builder.

Reviewer: ```suggestion
We recommend implementing your own `ReponseCalculator` if you need to save state between responses, for that you should implement your Response calculator following the above interface. The response calculation logic should go in the `ComputeResponse` method and you can set your initial state in a constructor.
```

- [Incredible Dot Product](https://github.com/Layr-Labs/eigensdk-go/blob/v2-dev-1/examples/incredible-dot-product/operator/main.go)
- [Awesome Vault Service](https://github.com/Layr-Labs/eigensdk-go/blob/v2-dev-1/examples/awesome-vault-service/operator/main.go)

## How to create your Response Calculator

To create your Response Calculator you have to declare a struct that satisfies the `ResponseCalculator` interface:

``` go
   type ResponseCalculator[Input any, Output any] interface {
      ComputeResponse(taskIndex uint32, input Input) (Output, error)
   }
```

The `ResponseCalculator` interface has the `ComputeResponse` method, which computes the response of a task given its index and input, returning the output, or an error if the computation fails.

> If you don't need to save a state between the operator responses you can use the `NewFunctionResponseCalculator` function, provided by the SDK in the operator package, that receives a function that computes the result from an input returning an output (or an error if fails) and returns a `ResponseCalculator` that on every call to the `ComputeResponse` method will call the received function.

Reviewer 2: Maybe we can remove the explanation about the `FunctionResponseCalculator` and just link to the documentation?

Author: Done in [13987f9](https://github.com/Layr-Labs/eigensdk-go/pull/700/commits/13987f94e8aae314477c9f05d567604fe04299ad)

If you need to save a state between the operator responses, then you should implement your Response calculator following the above interface.

> As a suggestion, you should place the calculation logic in the `ComputeResponse` method and initialize the necessary things for calculating the response on the new response calculator builder.

Reviewer 2: What if we explain this part first and then cover the case where the user doesn’t need to store state?

Author: Done in [7667d7e](https://github.com/Layr-Labs/eigensdk-go/pull/700/commits/7667d7efcf32fb3bbc7706cd500175fce9a1ff31) and [58d6a51](https://github.com/Layr-Labs/eigensdk-go/pull/700/commits/58d6a519f0dcb56557bf71ba3da5997192abc2cc)
